darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1173,8 +1173,7 @@ def predict_from_dataset(
         # Set mc_dropout rate
         self.model.set_mc_dropout(mc_dropout)
 
-        # setup trainer. will only be re-instantiated if both `trainer` and `self.trainer` are `None`
-        trainer = trainer if trainer is not None else self.trainer
```

Comment:
It'd be great if @dennisbader could take a look at this

Reply:
I'm discussing with @madtoinou about general improvements to how we use the PTL trainer. 

I am working on some changes which I will probably merge into this branch once ready.